patches/minecraft/net/minecraft/server/MinecraftServer.java.patch

```diff
@@ -9,39 +9,61 @@
           atomicreference.get().func_240802_v_();
        }, "Server thread");
        thread.setUncaughtExceptionHandler((p_240779_0_, p_240779_1_) -> {
-@@ -373,6 +373,7 @@
+@@ -373,10 +373,29 @@
              worldborder.func_177737_a(new IBorderListener.Impl(serverworld1.func_175723_af()));
              this.field_71305_c.put(registrykey1, serverworld1);
           }
 +         net.minecraftforge.common.MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.world.WorldEvent.Load(field_71305_c.get(registrykey)));
        }
  
     }
-@@ -384,6 +385,7 @@
++   
++   public void loadDimension(RegistryKey<World> registryKey, DimensionType type, ChunkGenerator gen) {
++       if(this.field_71305_c.get(registryKey) != null) {
++           return;
++       }
++       
++       IChunkStatusListener ichunkstatuslistener = this.field_213220_d.create(11);
```

Comment:
The IChunkStatusListener isn't really used except in loadInitialChunks which is why don't think it's necessary to keep a copy of it. I've already considered making it a NoopChunkStatusListener since it isn't used anyway, besides using memory.

Reply:
Umm... if you look in the `ChunkManager` class, the `IChunkStatusListener` is used every time the status of a chunk changes, like on load and unload. And if you look in the `Minecraft` class, it holds a reference to the latest `IChunkStatusListener` that is created from its factory, which makes it important to use the same instance as vanilla, otherwise vanilla's reference to its own `IChunkStatusListener` will be overwritten.

Reply:
Well, if you would actually look at the LoggingChunkStatusListener, which is the instance made by the factory, you would see that this IChunkStatusListener does nothing when not enabled. And the only instance where it is enabled is the loadInitialChunks method to log the chunk generation for the overworld. Therefore, it is not necessary to keep it.